src/unitxt/standard.py

```diff
@@ -50,12 +52,36 @@ def verify(self):
                 raise ValueError(
                     f"demos_pool_size must be bigger than num_demos={self.num_demos}, Got demos_pool_size={self.demos_pool_size}"
                 )
+            if self.loader_limit and self.demos_pool_size > self.loader_limit:
```

Comment:
you have if self.loader_limit in all of them. Do a single if on top of it all.

Also what if train_validatation > the limit for example?

Reply:
I'll move the if to the top.

self.max_validation_instances is checked below.